packages/application-components/src/components/modal-pages/custom-form-modal-page/custom-form-modal-page.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { MessageDescriptor } from 'react-intl';
+import buttonMessages from '../../../utils/button-messages';
+import ModalPage from '../internals/modal-page';
+import ModalPageHeader from '../internals/modal-page-header';
+import { ContentWrapper } from '../internals/modal-page.styles';
+import { Spacings } from '@commercetools-frontend/ui-kit';
+import {
+  FormPrimaryButton,
+  FormSecondaryButton,
+  FormDeleteButton,
+} from '../../../utils/default-form-buttons';
+
+type Label = string | MessageDescriptor;
+type Props = {
+  level?: number;
+  title: string;
+  isOpen: boolean;
+  onClose?: (event: React.SyntheticEvent) => void;
+  children: React.ReactNode;
+  zIndex?: number;
+  baseZIndex?: number;
+  getParentSelector?: () => HTMLElement;
+  shouldDelayOnClose?: boolean;
+  // TopBar Props
+  topBarCurrentPathLabel?: string;
+  topBarPreviousPathLabel?: Label;
+  // Header Props
+  subtitle?: string | React.ReactElement;
+  // Controls Props
+  controls: React.ReactNode;
```

Comment:
Nitpick: should we name this `formControls` maybe? We do that in the tabs form...